src/shared/lib/debounce.ts

```diff
@@ -0,0 +1,38 @@
+export default function debounce<AccumulatorType, ArgsType>(fn:(acc:AccumulatorType)=>void,
```

Comment:
I think the name debounce is too general for this thing.  It accumulates and then does something.
Also, using a closure, you do create an instance scope, but this is not a very clear pattern.  Instead, maybe we should make this a class, instantiate it in the cache and then give the class an add method.  
I'm also not totally sold that we need to libraritize this.  I think the cache having a timeout and then checking for its existence would be enough.  Could go either way on this one.  We should just be careful not to add too many "things" to our app.